api/v1alpha1/provisioning_types.go

```diff
@@ -143,6 +159,11 @@ type ProvisioningSpec struct {
 	// PXE deployments will always use the Provisioning Network and will not be
 	// affected by this flag.
 	VirtualMediaViaExternalNetwork bool `json:"virtualMediaViaExternalNetwork,omitempty"`
+
+	// PreprovisioningOSDownloadURLs set of boot Image URLs that would be necessary to provision a worker
+	// either using virtual media or PXE. The PXE URLs should be accessible from the
+	// provisioning network.
```

Comment:
I did not modify that comment. We should fix it in a different PR.

Reply:
This is new code in this PR.

Reply:
@zaneb I added this comment in https://github.com/openshift/cluster-baremetal-operator/pull/175.
I agree that the MachineOSDownloader should be able to pull them from any network.
The reason, I added this was:
As of today, we cannot PXE boot workers if they are not on the provisioning network. Does that still hold true for PXE booting via PreProvisioning PXE boot URLs? If so, would a better comment be "while using the Kernel, initramfs and rootfs URLs to PXE boot, the workers should be on the Provisioning network and that requirement does not exist if the workers are using the LiveISO URL to boot via virtualmedia".

Reply:
> @zaneb I added this comment in #175.
We're not planning to merge that though, right?

> As of today, we cannot PXE boot workers if they are not on the provisioning network. Does that still hold true for PXE booting via PreProvisioning PXE boot URLs?

Yes.

> If so, would a better comment be "while using the preprovisioning URLs to PXE boot, the workers should be on the Provisioning network and that requirement does not exist if the workers are using the LiveISO URL to boot via virtualmedia".

Even this is not strictly true, because if there is no LiveISO then the PXE files will be used for virtualmedia as well (with ironic building the ISO).

imho this is not a particularly useful context in which to remind people that PXE booting happens over the Provisioning network.

Reply:
> We're not planning to merge that separately though, right?
> 
No, I closed that PR.

Reply:
> imho this is not a particularly useful context in which to remind people that PXE booting happens over the Provisioning network.

Ageed. We'll have to be careful with our documentation given the user is presented with so many options with this new feature.